renderer/components/Editor.tsx

```diff
@@ -260,12 +269,68 @@ export class Editor extends React.Component<Props, State> {
     }
   }
 
+  shift = (number: number, shift: number) => {
+    return number * Math.pow(2, shift);
+  }
+
   toggleConsole = () => {
     this.props.toggleConsole();
     // Resize since the console overlaps with the editor, but enough time for console changes
     setTimeout(() => this.onWindowResize(), 0.01);
   }
 
+  // toggle keyboard control and add/remove listening for key presses to control robot
+  toggleKeyboardControl = () => {
+    
+    
+    this.setState({keyboardControl: !this.state.keyboardControl})
+    
+    if (!this.state.keyboardControl) {
+      window.addEventListener('keydown', this.turnCharacterOn, { passive: true});
+      window.addEventListener('keyup', this.turnCharacterOff, { passive: true});
```

Comment:
Let's add a short comment on what the `{ passive: true }` part does just to be a clearer